scripts/deploy_cluster_tar_single.sh

```diff
@@ -0,0 +1,174 @@
+            TYPE=$OPTARG
+            ;;
+        s)
+            ENABLE_SECURITY=$OPTARG
+            ;;
+        h)
+            usage
+            exit 1
+            ;;
+        :)
+            echo "-${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${OPTARG}"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate the required parameters to present
+if [ -z "$VERSION" ] || [ -z "$TYPE" ]; then
+  echo -e "\nERROR: You must specify '-v VERSION', '-t TYPE'"
+  usage
+  exit 1
+else
+  echo $VERSION $TYPE
+fi
+
+# Setup instances
+sudo sysctl -w vm.max_map_count=262144
+sudo chmod -R 777 /dev/shm
+ulimit -n 65535
+
+# Setup Work Directory
+cleanup
+DIR=$(mktemp --suffix=_INTEGTEST_WORKSPACE -d)
+mkdir -p $DIR/opensearch $DIR/opensearch-dashboards
+cp opensearch-onetime-setup.sh $DIR/opensearch
+cd $DIR; echo; echo New Deployment: $DIR
+
+# Download Artifacts
+echo -e "\nDownloading Artifacts Now"
+OPENSEARCH_URL="https://artifacts.opensearch.org/${TYPE}/bundle/opensearch/${VERSION}/opensearch-${VERSION}-linux-x64.tar.gz"
+DASHBOARDS_URL="https://artifacts.opensearch.org/${TYPE}/bundle/opensearch-dashboards/${VERSION}/opensearch-dashboards-${VERSION}-linux-x64.tar.gz"
+echo $OPENSEARCH_URL
+echo $DASHBOARDS_URL
+curl -s -f $OPENSEARCH_URL -o opensearch.tgz || exit 1
+curl -s -f $DASHBOARDS_URL -o opensearch-dashboards.tgz || exit 1
+ls $DIR
+
+# Extract Artifacts
+echo -e "\nExtract Artifacts Now"
+tar -xzf opensearch.tgz -C opensearch/ --strip-components=1
+tar -xzf opensearch-dashboards.tgz -C opensearch-dashboards/ --strip-components=1
+
+# Setup OpenSearch
+echo -e "\nSetup OpenSearch"
+cd $DIR/opensearch
+./opensearch-onetime-setup.sh
+mkdir -p backup_snapshots
+sed -i /^node.max_local_storage_nodes/d ./config/opensearch.yml
+echo "path.repo: [\"$PWD/backup_snapshots\"]" >> config/opensearch.yml
+echo "node.name: init-master" >> config/opensearch.yml
+echo "cluster.initial_master_nodes: [\"init-master\"]" >> config/opensearch.yml
+echo "cluster.name: opensearch-${VERSION}-linux-x64" >> config/opensearch.yml
+echo "network.host: 0.0.0.0" >> config/opensearch.yml
+echo "plugins.destination.host.deny_list: [\"10.0.0.0/8\", \"127.0.0.1\"]" >> config/opensearch.yml
+echo "script.context.field.max_compilations_rate: 1000/1m" >> config/opensearch.yml
+echo "webservice-bind-host = 0.0.0.0" >> plugins/opensearch-performance-analyzer/pa_config/performance-analyzer.properties
+
+# Start OpenSearch
+echo -e "\nStart OpenSearch"
+nohup ./opensearch-tar-install.sh > opensearch.log 2>&1 &
+
+# Setup Dashboards
+echo -e "\nSetup Dashboards"
+cd $DIR/opensearch-dashboards
+echo "server.host: 0.0.0.0" >> config/opensearch_dashboards.yml
+
+# Start Dashboards
+echo -e "\nStart Dashboards"
+cd $DIR/opensearch-dashboards/bin
+nohup ./opensearch-dashboards > opensearch-dashboards.log 2>&1 &
+
+# Wait for start
+echo -e "\nSleep 30"
+sleep 30
+echo Startup Complete
```

Comment:
It would be better to query the server every 5 seconds or so and look for a success response. Waiting 30 seconds doesn't guarantee that the server is up, and if the server comes up faster then we shouldn't be waiting around for it.

Reply:
Not really liking that since Dashboards specifically will return 200 even if it is not up.

Reply:
You'll need to look at the response content, not just the status code.